intermediate-python-debugging/exercise-using-debugger.md

```diff
@@ -0,0 +1,76 @@
+# Problem Set: Using the Debugger
+
+In this exercise we will use the debugger to identify two errors.  To run the activity clone the code at [AdaGold/python-debugging](https://github.com/AdaGold/python-debugging).  Then set up the project by creating a virtual environment and installing pytest.
+
+The code contains two bugs, one runtime error crashes the program on execution and the other is a logic error which calculates the incorrect sum for an Order instance.  
+
+## Activity
+
+### 1st Bug
+
+When you execute the program with:
+
+`$ python3 app.py`
+
+The program crashes with the following stack trace:
+
+```
+Traceback (most recent call last):
+  File "/Users/username/ada/python/python-debugging/app.py", line 16, in <module>
+    f"{calculate_total(kamala_order)}")
+  File "/Users/username/ada/python/python-debugging/grocery_store/order.py", line 16, in calculate_total
+    total += order["products"][i]["price"]
+TypeError: unsupported operand type(s) for +=: 'int' and 'str'
+```
+
+Hmmmm... It seems that it crashes at line 16 of `grocery_store/order.py`.
+
+Lets put a breakpoint at line 16 of `grocery_store/order.py`.
+
+![Visual of a breakpoint in order.py at line 16](../assets/vs-code-debugger/order-py-breakpoint.png)
+
+Then go to the debugger icon.
+
+![Debugger Icon](../assets/vs-code-debugger/debugger-icon.png)
+
+You **may** need to click on the select list, select `Add Configuration` and choose select `Python: Current File`.
+
+![select debug current python file](../assets/vs-code-debugger/debug-current-python-file.png)
+
+Otherwise you can simpily select "Python: Current File" from the drop down list.  
+
+### Running Debugger
+
+Then you can run the debugger by opening `app.py` and pressing the green triangle "Play" button.
+
+![Debugger Play Button](../assets/vs-code-debugger/start-debugger.png)
+
+You will notice that the application stops in `order.py`.
+
+![Running Debugger](../assets/vs-code-debugger/running-debugger.png)
+
+### Finding the bug...
+
+Notice that the loop attempts to add up all the product prices.  In Watch add a watch for `order["products"][i]`
+
+![Adding a watch on a product](../assets/vs-code-debugger/watch-price.png)
+
+
+<details style="max-width: 700px; margin: auto;">
+  <summary>
+    Do you see the problem?
+  </summary>
+
+  The price for the products gets set in `product.py`, see if you can find the error, fix it, and then re-run the debugger.  Does it crash now?  If you need add another breakpoint to `product.py` in the `create_product` function and step through the program.
+</details>
```

Comment:
Truthfully, I don't see a need to put this into a details/summary block... the answer in the details block doesn't seem to be "spoilable"? I don't think it detracts from the student to see as a paragraph underneath (not in a details block).